src/ir/datalog/program.h

```diff
@@ -89,6 +89,15 @@ class ArgumentType {
       : kind_(kind), name_(name) {}
   Kind kind() const { return kind_; }
   absl::string_view name() const { return name_; }
+  static ArgumentType MakePrincipalType() {
```

Comment:
Since we are making this part of the API, adding some tests would be good. 

Also, should you make the constructor private if you only expect this API to be used to create objects?

Reply:
done.